scripts/commit.sh

```
@@ -368,45 +919,57 @@ ${staged_with_tab}
-            git restore --staged $git_add
+            cleanup_on_exit "$git_add"
             exit
         fi
+        
+        # Sanitize commit message
+        if ! sanitize_commit_message "$commit_message"; then
+            show_sanitization_error "commit message" "Use printable characters only, 1-2000 characters."
+            cleanup_on_exit "$git_add"
+            exit 1
+        fi
+        commit_message="$sanitized_commit_message"
```

Shellcheck (0.10.0)

[warning] sanitized_commit_message is referenced but not assigned.

(SC2154)